web_src/js/index.js

```diff
@@ -2979,10 +2985,19 @@ function showDeletePopup() {
         return;
       }
 
-      $.post($this.data('url'), {
+      const postData = {
         _csrf: csrf,
-        id: $this.data('id')
-      }).done((data) => {
+      };
+      for (const key of Object.keys(dataArray)) {
```

Comment:
`of` is not very well supported on mobile browsers at least... Also in this case it would be better to use `in`:

```js
      for (const key in dataArray) {
```

Reply:
we use `of` in 35 places in `web_src/js` [1] and our linter configuration suggests not to use `in` at all - pushing us to use `of` - by setting: https://eslint.org/docs/rules/guard-for-in

(Whilst we could still fit get eslint to accept `for (x in y)` by using the recipes in that section `of` is idiomatic for the rest of our code.)

@silverwind do you have any thoughts?

[1]: ```bash
grep -r ' of ' web_src/js | grep -v '^[^:]*: *//' | wc -l
```

Reply:
In fact it's stronger than that our .eslintrc contains:

```eslintrc
  no-restricted-syntax: [2, WithStatement, ForInStatement, LabeledStatement]
```

Which means that `for (x in y) {` is explicitly disallowed. So I cannot use in.

Reply:
Asserted by #11615 which was approved by both of us.

Reply:
I've switched to use forEach from the Keys as per https://github.com/airbnb/javascript#iterators--nope

Reply:
for-of is fine and [very well supported](https://caniuse.com/?search=for...of), I think it's the clearly superiour alternative to `forEach` because it allows usage of flow control statements.

Reply:
Also see https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-for-each.md for a counter to that airbnb suggestion, which I think is weak.

Reply:
This really is heading in to bike-shedding. 

I've moved back to `of` as originally written - this matches the rest of our code in Gitea.